src/CMakeLists.txt

```diff
@@ -169,164 +157,45 @@ SET (HDRS
 	XMLParser.h
 )
 
-file(WRITE "${CMAKE_BINARY_DIR}/include/Globals.h"
-	"/* This file allows Globals.h to be included with an absolute path */\n#include \"${PROJECT_SOURCE_DIR}/Globals.h\"\n")
+set(FOLDERS
+	OSSupport HTTP Items Blocks Protocol Generating mbedTLS++ Bindings
+	WorldStorage Mobs Entities Simulator Simulator/IncrementalRedstoneSimulator
+	BlockEntities UI Noise
+)
+
+# Add all child source directories:
+foreach(folder ${FOLDERS})
+	add_subdirectory(${folder})
+endforeach(folder)
 
-include_directories("${CMAKE_BINARY_DIR}/include")
-include_directories (SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/../lib/TCLAP/include")
+# TODO: CMake includes PCH in every source file
+# Remove when every build has PCH enabled:
+file(WRITE "${CMAKE_BINARY_DIR}/include/Globals.h"
+	"/* This file allows Globals.h to be included with an absolute path */\n#include \"${PROJECT_SOURCE_DIR}/src/Globals.h\"\n")
 
 configure_file("BuildInfo.h.cmake" "${CMAKE_BINARY_DIR}/include/BuildInfo.h")
+target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE "${CMAKE_BINARY_DIR}/include/")
+
+# Enable PCH
+# TODO: upgrade CMake version and remove guard:
+if (PRECOMPILE_HEADERS AND CMAKE_VERSION VERSION_GREATER_EQUAL "3.16")
+	target_precompile_headers(${CMAKE_PROJECT_NAME} PRIVATE Globals.h)
+else()
+	message(WARNING "Precompiled headers for FASTER BUILDS not enabled, upgrade to CMake 1.16 or newer!")
+endif()
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+# Link process information library:
+if (WIN32)
+	target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE Psapi.lib)
```

Comment:
What happened to `ws2_32.lib`? Not needed?

Reply:
Linked by libevent, I think.